### PR TITLE
Maker: Led on() and off() stop Led animations

### PR DESCRIPTION
**Pull request by islemaster**
Makes `led.on()` and `led.off()` implicitly call `led.stop()`.  Before, if the led was blinking because you called `blink()`, the `on()` and `off()` methods would seem to have no effect because they wouldn't cancel the blink effect.  Now the led stops blinking when you call `on()` or `off()`.

[_Axosoft item_](https://codeorg.axosoft.com/viewitem?id=805&type=features&force_use_number=true).

My original approach was to extend `five.Led.RGB` similar to [how we customize some methods on the `Piezo` component](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/lib/kits/maker/Piezo.js) but this wasn't an option because the `blink()` method is [implemented using `on()` and `off()` (via `toggle()`) within johnny-five](https://github.com/code-dot-org/johnny-five/blob/master/lib/led/rgb.js#L446-L476) - so modifying those methods to _stop_ blinking broke `blink()`.  Instead our new `NeoPixel` controller wraps a `five.Led.RGB` controller instead of extending it - this requires we explicitly pass-through the usable interface, but gives correct behavior.

### CODE REVIEW

File: apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js [hunk 1]
```diff
@@ -125,126 +126,6 @@ describe('Circuit Playground Components', () => {
           });
         });
       });
-
-      // Note: The Mozilla color value documentation was very
-      // helpful when writing these tests:
-      // https://developer.mozilla.org/en-US/docs/Web/CSS/color_value
-      describe('that each accept color formats', () => {
```
Comment (islemaster) [inline apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js:132]: This set of tests moved into NeoPixelTest.

**Comment by islemaster** [comment]
@Hamms please review.  @epeach adding you as a reviewer as well, for any thoughts you have and because I may need your help working on this code in the near future.

Please note, one issue I'm _not_ addressing in this PR is that the same on/off/blink behavior change ought to be applied to the eleventh board LED (not a NeoPixel, can only be turned on and off) but it will require wrapping a different controller so I think it's worth doing in a separate PR.

File: apps/test/unit/lib/kits/maker/NeoPixelTest.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    it(`calls stop() on the led to end any animations`, () => {
+      led.off();
+      expect(five.Led.RGB.prototype.stop).to.have.been.calledOnce;
+    });
+
+    it(`calls the parent off() implementation`, () => {
+      led.off();
+      expect(five.Led.RGB.prototype.off).to.have.been.called;
+    });
+  });
```
Comment (Hamms) [inline apps/test/unit/lib/kits/maker/NeoPixelTest.js]: can we also test that blink does _not_ call stop?

**Comment by islemaster** [comment]
I didn't actually explore that.  I'd need to override `toggle()` (within johnny-five, `blink()` actually calls `toggle()` which calls `on()` and `off()`) to pass the extra parameter to on and off.  I guess that wouldn't require all the pass-through.  I'm not sure which I like better.

**Comment by islemaster** [comment]
It ended up being even cleaner than that, thanks to ES6 `super` calls.  PTAL.

**Comment by islemaster** [comment]
:( Might want to hold off on follow-up review, something in my changes broke some seemingly-unrelated unit tests.

**Pull request merged**